### PR TITLE
[reland][inductor] switch AotCodeCompiler to new cpp_builder

### PR DESCRIPTION
**Pull request by xuhancn**
Changes:
1. Switch `AotCodeCompiler` to new cpp_builder.
2. Only use `deprecated_cpp_compile_command` for `fb_code`, due to I can't debug anymore on no Meta internal environment access.
3. Add `TODO` comments for further some Meta employee help on contine to do this work.
4. Due to item 3, we only remaining `deprecated_cpp_compile_command` for `fb_code` to be fix, let's remove `validate_new_cpp_commands`.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: torch/_inductor/codecache.py
Comment (jansel) [inline torch/_inductor/codecache.py:1783]: Remove this line

File: torch/_inductor/codecache.py
Comment (jansel) [inline torch/_inductor/codecache.py:1796]: Remove old code?

File: torch/_inductor/codecache.py
Comment (jansel) [inline torch/_inductor/codecache.py:1886]: remove old code?

**Review by jansel — DISMISSED** [review]

**Review by jansel — APPROVED** [review]

**Comment by izaitsevfb** [comment]
@pytorchbot revert -m "breaks meta-internal tests" -c ghfirst

Apologies, but this change appears to be BC-breaking and needs to be imported manually.

This stacktrace is internal (and thus probably doesn't make much sense), but it shows that `.so` is not generated properly with our config:

```
.../deeplearning/aot_inductor/aot_inductor_lower_core.py:1109: AOTInductor successfully generated a .so at: c642u6kpus4hklyrimmyro5c3pm53qkkewscmdnxqhgbabkx23dm.so
.../deeplearning/aot_inductor/aot_inductor_lower_core.py:955: AOT_INDUCTOR_MODEL_NUM_RUNTIMES: 1
I0712 10:45:16.678714 30185 AOTInductorModelImpl.cpp:158] Loading .so lib from c642u6kpus4hklyrimmyro5c3pm53qkkewscmdnxqhgbabkx23dm.so onto device: cpu
ERROR

======================================================================
ERROR: test_cpu_lower_local_multi_embedding (deeplearning.aot_inductor.cpu.test.test_lowering_utils.CPULoweringTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../deeplearning/aot_inductor/cpu/test/test_lowering_utils.py", line 242, in test_cpu_lower_local_multi_embedding
    self._test_cpu_lower_local(lower_setting, model_kind)
  File ".../deeplearning/aot_inductor/cpu/test/test_lowering_utils.py", line 204, in _test_cpu_lower_local
    serialized_bytes, extra_files, modules_and_inputs = cpu_lower(
  File ".../deeplearning/aot_inductor/cpu/lowering_utils.py", line 248, in cpu_lower
    extra_files, lowered_module, lowered_module_sample_inputs = _cpu_lower(
  File ".../deeplearning/aot_inductor/cpu/lowering_utils.py", line 324, in _cpu_lower
    ) = generate_cpu_lowered_aot_inductor_module(
  File ".../deeplearning/aot_inductor/cpu/generate_cpu_lowered_aot_inductor_module.py", line 555, in generate_cpu_lowered_aot_inductor_module
    lowered = lower_func(
  File ".../deeplearning/aot_inductor/aot_inductor_lower_core.py", line 1134, in lower_pass
    torch.classes.fb.AOTInductorModel(*model_args),
RuntimeError: could not dlopen c642u6kpus4hklyrimmyro5c3pm53qkkewscmdnxqhgbabkx23dm.so: c642u6kpus4hklyrimmyro5c3pm53qkkewscmdnxqhgbabkx23dm.so: cannot open shared object file: No such file or directory

----------------------------------------------------------------------
Ran 1 test in 23.364s
```

**Comment by clee2000** [comment]
I'm pretty sure the dataloader segfault failure is unrelated, [there was a disable issue](https://togithub.com/pytorch/pytorch/issues/128551) for it that recently got closed which I have now reopened

**Comment by atalman** [comment]
thank you @clee2000 relanding it now

**Comment by clee2000** [comment]
@pytorchbot revert -m "broke internal tests D60265910" -c ghfirst

Here's the error logs, it looks like it's still facing the same problem?
```
======================================================================
ERROR: test_add_complex_abi_compatible_cpu_with_stack_allocation (caffe2.test.inductor.test_aot_inductor.AOTInductorTestABICompatibleCpuWithStackAllocation)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test/inductor/test_torchinductor.py", line 11128, in new_test
    return value(self)
  File "test/inductor/test_aot_inductor.py", line 2948, in test_add_complex
    self.check_model(Model(), (x, y))
  File "test/inductor/test_aot_inductor.py", line 115, in check_model
    actual = AOTIRunnerUtil.run(
  File "test/inductor/test_aot_inductor_utils.py", line 115, in run
    optimized = AOTIRunnerUtil.load(device, so_path)
  File "test/inductor/test_aot_inductor_utils.py", line 83, in load
    runner = AOTIRunnerUtil.load_runner(device, so_path)
  File "test/inductor/test_aot_inductor_utils.py", line 69, in load_runner
    return test_aot_inductor_model_runner_pybind.Runner(
RuntimeError: Failure loading .so: cksb2px2uh2ogvn3gsjwhtez5mf433e7pla7rtxw4gdd2p3d735j.so: cannot open shared object file: No such file or directory

----------------------------------------------------------------------
Ran 1 test in 6.217s
```
@jansel Could you help provide any additional information?

**Comment by clee2000** [comment]
@desertfire @angelayi Can you help out with this?  Jason is on PTO

**Comment by xuhancn** [comment]
> @desertfire @angelayi Can you help out with this? Jason is on PTO

I will revert some code to origin code, and make sure it could pass `fb_code`.

**Comment by ZainRizvi** [comment]
@pytorchbot revert -c ghfirst -m "broke internal tests"

Sorry but this still is having similar failures. @desertfire @angelayi might you folks be able to help while @jansel is on PTO?

```
======================================================================
ERROR: test_aot_batch_accuracy (accelerators.workloads.models.slimdsnn.test_slimdsnn.SlimDSNN)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/xarfuse/uid-30041/63ae44bd-seed-nspid4026533646_cgpid2373577-ns-4026533643/accelerators/workloads/models/slimdsnn/test_slimdsnn.py", line 24, in test_aot_batch_accuracy
    _aot(
  File "/mnt/xarfuse/uid-30041/63ae44bd-seed-nspid4026533646_cgpid2373577-ns-4026533643/accelerators/workloads/models/slimdsnn/slimdsnn.py", line 179, in _aot
    aot_lower(
  File "/mnt/xarfuse/uid-30041/63ae44bd-seed-nspid4026533646_cgpid2373577-ns-4026533643/accelerators/workloads/models/slimdsnn/slimdsnn.py", line 233, in aot_lower
    ) = generate_cpu_lowered_aot_inductor_module(
  File "/mnt/xarfuse/uid-30041/63ae44bd-seed-nspid4026533646_cgpid2373577-ns-4026533643/deeplearning/aot_inductor/cpu/generate_cpu_lowered_aot_inductor_module.py", line 616, in generate_cpu_lowered_aot_inductor_module
    lowered = lower_func(
  File "/mnt/xarfuse/uid-30041/63ae44bd-seed-nspid4026533646_cgpid2373577-ns-4026533643/deeplearning/aot_inductor/aot_inductor_lower_core.py", line 1181, in lower_pass
    torch.classes.fb.AOTInductorModel(*model_args),
RuntimeError: could not dlopen c6hx23jl34eyphwv52hlzh4bl7sedyv6lrufvxjno7qapoqzadkc.so: c6hx23jl34eyphwv52hlzh4bl7sedyv6lrufvxjno7qapoqzadkc.so: cannot open shared object file: No such file or directory
```

**Comment by desertfire** [comment]
@xuhancn , I thought you are going to land https://github.com/pytorch/pytorch/pull/131304 instead? And as we discussed offline, https://github.com/pytorch/pytorch/pull/131304 still hits the same test failure after imported. You need to fix that before landing.

**Comment by ZainRizvi** [comment]
(retrying the revert)

**Comment by xuhancn** [comment]
> @xuhancn , I thought you are going to land #131304 instead? And as we discussed offline, #131304 still hits the same test failure after imported. You need to fix that before landing.

@desertfire, I can't land the `AotCodeCompiler` without Meta internal access. I checked your provided error info, it is not enough to fix issue. 
I think need some Meta engineer help on land `AotCodeCompiler`.
I just want to remove the `_do_validate_cpp_commands`, due to it is break other builder related development.

**Comment by xuhancn** [comment]
close by https://github.com/pytorch/pytorch/pull/132766

**Pull request closed without merging**
